api_v3/services/ThumbAssetService.php

```diff
@@ -949,6 +952,21 @@ public function addFromImageAction($entryId, $fileData)
 		return $thumbAssets;
 	}
 	
+	protected function isValidFileType($fileName, $partnerId = null)
+	{
+		if(!$partnerId)
+		{
+			$partnerId = kCurrentContext::getCurrentPartnerId();
+		}
+		$fileExtension = strtolower(pathinfo($fileName, PATHINFO_EXTENSION));
+		if (PermissionPeer::isValidForPartner(PermissionName::FEATURE_FILE_TYPE_RESTRICTION_PERMISSION, $partnerId)
+			&& in_array($fileExtension, $this->restrictedThumbnailFileTypes))
```

Comment:
move the array from the const inside the function, it doesnt really have use outside of it